### PR TITLE
AGR-2590 ansible devops part2 - Split out DOCKER_BUILD_TAG and DOCKER_PULL_TAG

### PR DESCRIPTION
**Pull request by mluypaert**
Please review and approve (once fit), but do not merge yet (I will do so after approval, once I updated the necessary GoCD pipelines, one pipeline at a time).

### CODE REVIEW

File: tasks/push_cacher_image.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     state: stopped
 
 - name: Commit container
-  command: docker commit "{{ INFINISPAN_SERVER_NAME }}" "{{ CACHER_RELEASE_IMAGE_NAME }}"
+  command: docker commit "{{ INFINISPAN_SERVER_NAME }}" "{{ CACHER_RELEASE_REG_INAME }}:{{ DOCKER_BUILD_TAG }}"
 
 - name: Push Container To Docker Hub
-  command: docker push "{{ CACHER_RELEASE_IMAGE_NAME }}"
+  command: docker push "{{ CACHER_RELEASE_REG_INAME }}:{{ DOCKER_BUILD_TAG }}"
```
Comment (oblodgett) [inline tasks/push_cacher_image.yml:10]: This shouldn't need the `:{{ DOCKER_BUILD_TAG }}`
it is being set in the env
`CACHER_RELEASE_IMAGE_NAME: "{{ CACHER_RELEASE_REG_INAME }}:{{ DOCKER_PULL_TAG }}"`

Comment (oblodgett) [reply tasks/push_cacher_image.yml:10]: or switch it back to `CACHER_RELEASE_IMAGE_NAME`

File: tasks/push_neo_image.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@
     state: stopped
 
 - name: Commit container
-  command: docker commit "{{ NEO_SERVER_NAME }}" "{{ NEO_RELEASE_IMAGE_NAME }}"
+  command: docker commit "{{ NEO_SERVER_NAME }}" "{{ NEO_RELEASE_REG_INAME }}:{{ DOCKER_BUILD_TAG }}"
 
 - name: Push Container
-  command: docker push "{{ NEO_RELEASE_IMAGE_NAME }}"
+  command: docker push "{{ NEO_RELEASE_REG_INAME }}:{{ DOCKER_BUILD_TAG }}"
```
Comment (oblodgett) [inline tasks/push_neo_image.yml:11]: See previous comment.

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: tasks/push_cacher_image.yml (hunk 1 above)
Comment (mluypaert) [reply tasks/push_cacher_image.yml]: As per our discussion on slack: `DOCKER_BUILD_TAG`  != `DOCKER_PULL_TAG`. I did move all image variable definitions into [shared_variables.yml](./environments/shared_variables.yml) now.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**